plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/cache/MutableDeltaLogDeltaLakeCacheKeyProvider.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.cache;
+
+import io.trino.filesystem.TrinoInputFile;
+
+import java.util.Optional;
+
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogUtil.TRANSACTION_LOG_DIRECTORY;
+
+public class MutableDeltaLogDeltaLakeCacheKeyProvider
```

Comment:
On second thought, could we instead combine this logic into the existing DeltaLakeCacheKeyProvider (so inject the configuration in the constructor and store the value of getDeltaLogCacheEnabled). Sorry for the misdirection, but this probably does not warrant a separate class.

Reply:
Agreed!